src/Centeva.ObjectStorage.Azure/AzureObjectStorage.cs

```diff
@@ -3,33 +3,29 @@
 using Azure.Storage.Blobs.Models;
 using Azure.Storage.Blobs.Specialized;
 using Azure.Storage.Sas;
+using Centeva.ObjectStorage.Connections;
 
 namespace Centeva.ObjectStorage.Azure;
 
 public class AzureObjectStorage : ISignedUrlObjectStorage
 {
     private readonly BlobServiceClient _client;
-    private readonly AzureObjectStorageConnectionString _connectionString;
-    private readonly string _containerName;
+    private readonly ObjectStorageConnectionString _connectionString;
+    private string? _containerName = null;
 
-    public AzureObjectStorage(string containerName, AzureObjectStorageConnectionString? connectionString, string accountName, string accountKey)
+    public AzureObjectStorage(ObjectStorageConnectionString? connectionString)
```

Comment:
The other providers (Google, S3) deliberately did not allow the `ObjectStorageConnectionString` to be passed into their constructors.  Parsing the checking for required components was done in the factory classes.  Was there something preventing that from happening here?